src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -7,16 +7,20 @@
 
 
 @pytest.fixture(scope='function')
-def device_info(request):
+def device_info(request, name):
     device_info = {}
-    device_name = "unknown"
+    device_name = name
     device_sn = "unknown"
     try:
         with nimodinst.Session('nidmm') as session:
             if len(session) > 0:
-                device_name = session.device_name[0]
-                device_sn = session.serial_number[0]
-
+                for x in range(0, len(session)):
+                    if (session.device_name[x] == device_name):
+                        device_sn = session.serial_number[x]
+                        break
+                if (device_sn == "unknown"):
```

Comment:
Instead of checking after the fact to see whether to use information from modinst or not instead do something like this:
```
with nimodinst.Session('nidmm') as session:
    device_name = None
    if len(session) > 0:
        device_name = session.device_name[0]
    for device in session:
        if device.device_name == name
            device_name = name
    assert device_name is not None, "No NI-DMM HW installed"
```